### PR TITLE
Remove future_stencil

### PR DESCRIPTION
**Pull request by jdahm**
## Purpose

Removes the `FutureStencil` class and `future_stencil` entry-point function.

## Code changes:

- Removes the `FutureStencil` driver for gt4py stencils.
- Removes its tests.

### CODE REVIEW

File: dsl/pace/dsl/stencil.py [hunk 1]
```diff
@@ -298,21 +296,10 @@ def __init__(
...
-        # Enable distributed compilation if running in parallel
-        if MPI is not None and MPI.COMM_WORLD.Get_size() > 1:
-            stencil_function = future_stencil.future_stencil
-            stencil_kwargs["wrapper"] = self
-        else:
-            # future stencil provides this information and
-            # we want to be consistent with the naming whether we are
-            # running in parallel or not (so we use the same cache)
-            stencil_kwargs["name"] = func.__module__ + "." + func.__name__
+        # stencil_kwargs["name"] = func.__module__ + "." + func.__name__
```
Comment (elynnwu) [inline dsl/pace/dsl/stencil.py:299]: I think you want to uncomment this

**Review by elynnwu — APPROVED** [review]

File: dsl/pace/dsl/stencil.py [hunk 2]
```diff
@@ -224,6 +229,7 @@ def __init__(
         )
         numpy_stencil_config = StencilConfig(
             backend="gtc:numpy",
+            cache_dir=".gt_cache",
```
Comment (elynnwu) [inline dsl/pace/dsl/stencil.py:232]: Wouldn't this have to match with the rank as well? This is supposed to check if numpy give you the same answer.

File: driver/examples/configs/baroclinic_c12.yaml [hunk 3]
```diff
@@ -1,5 +1,6 @@
 stencil_config:
   backend: gtc:numpy
+  cache_dir: .gt_cache_{rank}
```
Comment (elynnwu) [inline driver/examples/configs/baroclinic_c12.yaml:3]: Do we expect the user to change this?

File: dsl/pace/dsl/stencil.py [hunk 4]
```diff
@@ -804,14 +795,20 @@ def get_quantity_halo_spec(
...
             grid_indexing: configuration for domain and halo indexing`
         """
         self.config: StencilConfig = config
         self.grid_indexing: GridIndexing = grid_indexing
+        self.comm: Comm = comm
+
+        # Replace {rank} in self.config.cache_dir with the communicator's rank
+        self.config.cache_dir = self.config.cache_dir.replace(
+            "{rank}", str(comm.Get_rank())
+        )
```
Comment (elynnwu) [inline dsl/pace/dsl/stencil.py:811]: Is the idea here to allow 6 ranks to use one single cache as well? unless specified by the namelist to have 6 separate caches?

File: dsl/pace/dsl/stencil.py (hunk 2 above)
Comment (jdahm) [reply dsl/pace/dsl/stencil.py:232]: Ah, right. This should probably match the one in the `stencil_config` passed for consistency!

File: dsl/pace/dsl/stencil.py (hunk 4 above)
Comment (jdahm) [reply dsl/pace/dsl/stencil.py:812]: Yeah, the user can basically select programmatically, or through the driver yaml config.

File: dsl/pace/dsl/stencil.py (hunk 1 above)
Comment (jdahm) [reply dsl/pace/dsl/stencil.py:307]: I checked and I think the default is different, so yeah I'll change it back. Thanks!

**Comment by jdahm** [comment]
It turns out that this did not jive well with the way `dace` requires it done for whole-program optimization.

**Pull request closed without merging**